src/main/java/com/synopsys/integration/jenkins/scan/service/scan/blackduck/BlackDuckParametersService.java

```diff
@@ -93,6 +95,11 @@ public BlackDuck prepareBlackDuckObjectForBridge(Map<String, Object> blackDuckPa
                         blackDuck.getAutomation().setPrComment(Boolean.parseBoolean(value));
                     }
                     break;
+                case ApplicationConstants.BRIDGE_BLACKDUCK_DOWNLOAD_URL_KEY:
+                    Download download = new Download();
```

Comment:
Ideally this statement should be inside the `Blackduck() `constructor so that entire code base is consistent.

Reply:
As downloa_url is optional param. If param is not provided, unnecessary an object will be created.

Reply:
And all the BridgeInput's members' object are created on requirement.